### DISCUSSION TITLE
Query role name by where like

### DISCUSSION THREAD

**Discussion by montanhes** [spatie/laravel-permission#1987] [Q&A] [answered]
How can I query the Users models that has a role using the where like?
I mean, something like: 

```
$partialString = 'adm';
User::role()->where('name', 'like', '%'.$partialString.'%')->get();
```

**Comment by montanhes** [comment, marked as answer]
Oh I found an issue #985 with the solution. I was trying to search for roles on a datatable.

I just added the relationship and got it to work like all others relationships:

`User::with('roles')`

**Comment by erikn69** [comment]
```php
User::whereHas('roles', fn($query)  => $query->where('name', 'like', '%'.$partialString.'%'))->get();
```
